definitions/npm/react-i18next_v5.x.x/flow_v0.53.x-/react-i18next_v5.x.x.js

```diff
@@ -0,0 +1,46 @@
+declare module "react-i18next" {
+  declare type TFunction = (
+    key?: ?string,
+    data?: ?{ [key: string]: string }
+  ) => string;
+  declare type Locales = string | Array<string>;
+
+  declare type TranslatorProps = {
+    t: TFunction,
+    i18nLoadedAt: ?Date,
+    i18n: Object
+  };
+
+  declare type StatelessComponent<P> = (props: P) => ?React$Element<any>;
+  declare type LegacyTranslator<OP, P> = {
+    (component: StatelessComponent<P>): Class<React$Component<OP, void>>,
+    <St>(
+      component: Class<React$Component<P, St>>
+    ): Class<React$Component<OP, St>>
+  };
```

Comment:
For some reason, the old way of writing this does work, but this does not:

```js
  declare type LegacyTranslator<OP, P> = (
    component: React$ComponentType<P>
  ) => Class<React$Component<OP, *>>
```

The error:

![screen shot 2017-08-23 at 5 38 45 pm](https://user-images.githubusercontent.com/6344219/29641608-f3389aee-8829-11e7-9bf8-c3f2826e8729.png)